assign0/README.md

```diff
@@ -0,0 +1,354 @@
+
+### Hints
+
+You can check that an element is contained in a collection by using the keyword `in`.
+
+### Submission
+
+Please submit a file called `cheese.py` which contains the code for this segment of the assignment.
+
+## Part 3: Chatbot
+
+In the third part of this assignment, you will implement a chatbot program which has a conversation with the user. There are just two requirements for this program:
+
+1. *Ask the user for input and print out responses*. The program should read input from the console and, based on the input, print out some response. You don't have to respond immediately after every input; the program could keep asking questions, just as long as it responds eventually.
+2. *Access stored data*. The program should access data stored in a text file over the course of its run, so that data from previous runs is accessible to users in future runs.
+
+To give you an idea of the scope and complexity of what we expect from this part of the assignment, below are two examples of types of submissions which would fulfill the requirements of this component.
+
+### Example 1: The Doors of Destiny
+
+A simple user authentication chatbot might take the role of a gatekeeper. The gatekeeper stops all travellers attempting to pass through the gate, and asks them for their name and passphrase. If the traveller provides a name and passphrase whch are contained in the gatekeeer's Book of Records, the traveller may proceed. If not, the traveller is denied entry.
+
+Upon being asked for a name and passphrase, each traveller has two options: they may (a) provide a valid name and passphrase, or (b) bribe the guard - which allows them to add a new name and passphrase to the Book of Records - for a small fee.
+
+A sample run for such a chatbot might appear as follows (user input is ***bolded and italicized***):
+
+<pre><code>
+(cs41-env)$ python chatbot.py
+Halt! Welcome to the Doors of Destiny. 
+Should you wish to proceed, you must identify yourself within the Book of Records. 
+
+Is your name present in our book? <b><i>yes</i></b>
+What is your name, traveller? <b><i>Michael</i></b>
+What is your passphrase? <b><i>parthsarin12345</i></b>
+Welcome through, peaceful soul!
+
+(cs41-env)$ python chatbot.py
+Halt! Welcome to the Doors of Destiny. 
+Should you wish to proceed, you must identify yourself within the Book of Records. 
+
+Is your name present in our book? no
+Psst! I'm... not supposed to tell you this, but for a small... compensation... I might be able to add you to the Book of Records without the Warden noticing. 
+
+Would you like to be added to the Book of Records? <b><i>yes</i></b>
+Perfect! I've added you - but I don't come cheap! I charge 100 Dogecoin for my services. 
+Can you make the deposit? <b><i>yes</i></b>
+Deposit successful! You have 5532 Dogecoin remaining in your account.
```

Comment:
@cooper-mj This kind of functionality is really tricky, no? Changing just one piece of the file in place? You kinda have to copy the entire file, make a change to just one line (which is already tricky - you shouldn't use the replace method on a string, e.g.) and then write that over the existing file, I think.

Reply:
Excellent point - you're right that keeping everything in the same file would make things probably unnecessarily tricky. Upon second thought, it seems that your suggestion below - of tracking the Dogecoin in one file, and the usernames/passwords in another file - would be a better way to go on this. I'll revise and make a commit accordingly.